profiles/management/commands/gen_realistic_users.py

```diff
@@ -0,0 +1,256 @@
+"""
+Generates a set of realistic users to help us test search functionality
+"""
+from datetime import datetime, timedelta
+from django.core.management import BaseCommand
+from django.contrib.auth.models import User
+from profiles.models import Employment, Education
+from courses.models import Program, Course, CourseRun
+from dashboard.models import CachedCertificate, CachedEnrollment
+from micromasters.utils import load_json_from_file
+
+
+USER_DATA_PATH = 'profiles/management/realistic_user_data.json'
+PROGRAM_DATA_PATH = 'profiles/management/realistic_program_data.json'
+FAKE_USER_USERNAME_PREFIX = 'fake.'
+FAKE_PROGRAM_DESC_PREFIX = '[FAKE] '
+CACHED_MODEL_LAST_REQUEST = datetime.now() + timedelta(days=365)
+
+
+# Util functions
+
+def first_matching_item(iterable, predicate):
+    """Returns the first item in an iterable that matches a given predicate"""
+    return next(x for x in iterable if predicate(x))
+
+
+def filter_dict_by_key_set(dict_to_filter, key_set):
+    """
+    Takes a dictionary and returns the same dictionary without keys that don't exist in a given set
+    """
+    return {key: dict_to_filter[key] for key in dict_to_filter.keys() if key in key_set}
+
+
+def get_non_relation_field_names(model_cls):
+    """
+    Gets a set of field names from a model class that aren't relation fields
+    """
+    return set(field.name for field in model_cls._meta.get_fields() if not field.is_relation) # pylint: disable=protected-access
+
+
+def deserialize_model_data_on_object(model_obj, data, save=True):
+    """
+    Sets field values on an existing model object using some supplied data
+    """
+    non_relation_field_names = get_non_relation_field_names(model_obj.__class__)
+    for k, v in filter_dict_by_key_set(data, non_relation_field_names).items():
+        setattr(model_obj, k, v)
+    if save:
+        model_obj.save()
+    return model_obj
+
+
+def deserialize_model_data(model_cls, data, relational_data=None):
+    """
+    Creates a new instance of a model class and fills in field values using some supplied data
+    """
+    non_relation_field_names = get_non_relation_field_names(model_cls)
+    model_data = filter_dict_by_key_set(data, non_relation_field_names)
+    if relational_data:
+        model_data.update(relational_data)
+    return model_cls.objects.create(**model_data)
```

Comment:
@aliceriot i split this deserialization into 2 functions. `deserialize_model_data_on_object` is only needed for deserializing profiles. we need this b/c when a User is created, a Profile object is automatically created as well, so i needed a way to deserialize data onto that existing object. changed everything else to use `deserialize_model_data` and pass in a class